File: ABC/ABC301-400/ABC338/E/accumulate2/E.nim
import strutils, sequtils

let N = stdin.readLine.parseInt
var
  A = newSeq[int](N)
  B = newSeq[int](N)
  cnt = newSeq[int](2 * N)
for i in 0..<N:
  let a_b = stdin.readLine.split().map(parseInt)
  A[i] = a_b[0] - 1
  B[i] = a_b[1] - 1
  if A[i] > B[i]:
    swap(A[i], B[i])
  inc cnt[A[i]]
  dec cnt[B[i]]
for i in 1..<(2 * N):
  cnt[i] += cnt[i - 1]
for i in 0..<N:
  if cnt[A[i]] < cnt[B[i] - 1]:
    echo "Yes"
    quit()
echo "No"

File: ABC/ABC301-400/ABC338/E/accumulate2/E.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int N;
	cin >> N;
	vector<int> A(N), B(N), cnt(2 * N);
	for (int i = 0; i < N; i++) {
		cin >> A[i] >> B[i];
		A[i]--; B[i]--;
		if (A[i] > B[i]) {
			swap(A[i], B[i]);
		}
		cnt[A[i]]++;
		cnt[B[i]]--;
	}
	for (int i = 1; i < 2 * N; i++) {
		cnt[i] += cnt[i - 1];
	}
	for (int i = 0; i < N; i++) {
		if (cnt[A[i]] < cnt[B[i] - 1]) {
			cout << "Yes\n";
			return 0;
		}
	}
	cout << "No\n";
}